### PR TITLE
Update tuya.ts

### PR DESCRIPTION
**Pull request by oroGithub**
Added new device:
'_TZE204_dapwryy7'
('5.8 GHz human presence sensor')

### CODE REVIEW

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -10651,6 +10651,120 @@ const definitions: Definition[] = [
...
+            e
+                .numeric('small_move_sensitivity', ea.STATE_SET)
+                .withValueMin(0)
+                .withValueMax(10)
+                .withValueStep(1)
+                .withDescription('Small Move sensitivity'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [
```
Comment (Koenkk) [inline src/devices/tuya.ts:10741]: Could you also return a `presence` here (a boolean) and add `e.presence()` to `exposes`? Example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/0f0917687d962efcca7788ac392d32fe76afdc16/src/devices/onenuo.ts#L30

File: src/devices/tuya.ts [hunk 2]
```diff
@@ -10651,6 +10651,120 @@ const definitions: Definition[] = [
...
+        model: 'ZG-205Z',
+        vendor: 'Tuya',
+        description: '5.8 GHz human presence sensor',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        onEvent: tuya.onEventSetTime,
+        exposes: [
+            e
+                .enum('presence_state', ea.STATE, ['none', 'presence', 'peaceful', 'small movement', 'large movement'])
+                .withDescription('\'none\', \'peaceful\' or \'large movement\''),
```
Comment (Koenkk) [inline src/devices/tuya.ts:10665]:
```suggestion
                .withDescription('The presence state'),
```

File: src/devices/tuya.ts (hunk 1 above)
Comment (oroGithub) [reply src/devices/tuya.ts:10741]: Done and tested :)
Now I have to figure out how to get it in here haha. Let's try this
```suggestion
                [
                    1,
                    null,
                    {
                        from: (v) => {
                            const lookup = {none: tuya.enum(0), presence: tuya.enum(1), peaceful: tuya.enum(2), 'small movement': tuya.enum(3), 'large movement': tuya.enum(4)};
                            const presenceState = Object.entries(lookup).find((i) => i[1].valueOf() === v)[0];
                            return {
                                presence: presenceState != 'none',
                                presence_state: presenceState,
                            };
                        },
                    },
                ],
```

File: src/devices/tuya.ts [hunk 3]
```diff
@@ -10651,6 +10651,120 @@ const definitions: Definition[] = [
...
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_dapwryy7']),
+        model: 'ZG-205Z',
+        vendor: 'Tuya',
+        description: '5.8 GHz human presence sensor',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        onEvent: tuya.onEventSetTime,
+        exposes: [
+            e
```
Comment (oroGithub) [inline src/devices/tuya.ts:10663]:
```suggestion
            e.presence(),
            e
```

File: src/devices/tuya.ts [hunk 4]
```diff
@@ -10651,6 +10651,120 @@ const definitions: Definition[] = [
...
+                    1,
+                    'presence_state',
+                    tuya.valueConverterBasic.lookup({
+                        'none': tuya.enum(0),
+                        'presence': tuya.enum(1),
+                        'peaceful': tuya.enum(2),
+                        'small movement': tuya.enum(3),
+                        'large movement': tuya.enum(4),
+                    }),
+                ],
```
Comment (oroGithub) [inline src/devices/tuya.ts:10751]:
```suggestion
```

**Review by oroGithub — COMMENTED** [review]
Suggestions implemented-

**Comment by Koenkk** [comment]
Could you update your code with these changes/

**Comment by oroGithub** [comment]
> Could you update your code with these changes?

Done ... I think. I've lost track of things a bit.

**Comment by Koenkk** [comment]
thanks!

**Pull request merged**